sdk/core/Azure.Core/tests/TestFramework/TestRecording.cs

```diff
@@ -210,6 +210,16 @@ public string GenerateId()
         {
             return Random.Next().ToString();
         }
+		
+		public string GenerateId(int maxValue)
```

Comment:
I chose to add override methods instead of int? as it has less logic and less prone to error but let me know if you prefer a different implementation. Our service needs this and I thought it was easy enough to just make a PR for it.